src/main/scala/io/iohk/ethereum/domain/branch/Branch.scala

```diff
@@ -2,17 +2,8 @@ package io.iohk.ethereum.domain.branch
 
 import akka.util.ByteString
 
-import io.iohk.ethereum.domain.Block
+sealed trait Branch
 
-/** An interface to manipulate blockchain branches */
-trait Branch {
+case class BestBranchSubset(tipBlockHash: ByteString, tipBlockNumber: BigInt) extends Branch
```

Comment:
Why did you add the suffix `Subset` here?

Reply:
It works with any subset on the best branch so I wanted to make it clear and kept the name in from my initial draft of the case class representation. The idea was that we could use it later to define a fork as a tip and a best branch subset which is common to the fork and our current best branch. 

But I agree that it might be confusing and not really relevant to our use case for now. I'll just name it BestBranch as before